packages/react-app/src/sections/Header.tsx

```diff
@@ -70,12 +73,26 @@ function Header({ isDarkMode, darkModeToggle }) {
     }
   }
 
+async function handleChangeNetwork(){
+const chainIds = envConfig.allowedNetworks;
+  if (window.ethereum.networkVersion !== chainIds) {
```

Comment:
Here, chainIds is an array. Try checking if chainIds array includes the currently connected network in the Metamask, if not then switch to the coreContractChain chainId from the envConfig.